src/components/Legend/helper.js

```diff
@@ -0,0 +1,33 @@
+// Helper function to group elements by segment and ring
+export const groupElements = (elements, segments, rings) => {
+  const grouped = {};
+
+  // Initialize the grouped object with segments and rings
+  segments.forEach((segment) => {
+    grouped[segment.slug] = {
+      label: segment.label,
+      color: segment.color,
+      rings: {},
+    };
+
+    rings.forEach((ring) => {
+      grouped[segment.slug].rings[ring.slug] = {
+        label: ring.label,
+        elements: [],
+      };
+    });
+  });
+
+  // Group elements by segment and ring
+  elements.forEach((element, index) => {
+    const segment = grouped[element.segment];
```

Comment:
maybe we can use just const ring = grouped[element.segment]?.rings[element.ring]; ?